fenics_ice/solver.py

```diff
@@ -1077,6 +1106,50 @@ def bglen_data_conditional(self, H, glen_mask, H_float=None):
 
         return fl_beta_mask
 
+    def melt_conditional(self, H, H_DG):
+        """Compute a ufl Conditional where melt nonzero=1, no melt=0"""
+
+        constants = self.params.constants
+        rhow = constants.rhow
+        rhoi = constants.rhoi
+        H_float_DG = -(rhow/rhoi) * self.bed_DG
+
+        # Note: cell=triangle just suppresses a UFL warning ("missing cell")
+        melt_mask = ufl.operators.Conditional(ufl.operators.And(H_DG < H_float_DG, 
+                                              ufl.operators.And(self.model.melt_domains > 0, H > 10.0)),
+                                          Constant(1.0, cell=triangle, name="Const Melt"),
+                                          Constant(0.0, cell=triangle, name="Const No Melt"))
+
+        return melt_mask
+
+    def melt_depth_conditional(self):
+        """Compute a ufl Conditional where melt domain = 1 or 2, and return appropriate param"""
+
+        constants = self.params.melt
+        depth1 = constants.depth_therm_domain_1
+        depth2 = constants.depth_therm_domain_2
+
+        # Note: cell=triangle just suppresses a UFL warning ("missing cell")
+        melt_depth = ufl.operators.Conditional(ufl.eq(self.model.melt_domains,1.0),
+                                          Constant(depth1, cell=triangle, name="Depth Domain 1"),
+                                          Constant(depth2, cell=triangle, name="Depth Domain 2"))
```

Comment:
Could use `(self.model.melt_domains - 1) * depth2 - (self.model.melt_domains - 2) * depth1` to avoid the conditional -- although might then want to add a check that `self.model.melt_domains` equals 1 or 2 everywhere

Reply:
i think this is resolved.. got rid of these conditionals